webview-ui/src/components/settings/ApiOptions.tsx

```diff
@@ -507,7 +513,7 @@ const ApiOptions = ({
 				</div>
 			)}
 
-			{apiConfiguration?.apiProvider === "vertex" && (
+			{selectedProvider === "vertex" && (
```

Comment:
I couldn't figure out why this was the only check like this... everything else uses `selectedProvider`.